src/core/functions/internal_functions/system/MultiSelectModal.ts

```diff
@@ -0,0 +1,229 @@
+import { Modal, TextComponent, fuzzySearch, prepareQuery } from "obsidian";
+import { TemplaterError } from "utils/Error";
+
+export class MultiSelectModal<T> extends Modal {
+    private resolve: (values: T[]) => void;
+    private reject: (reason?: TemplaterError) => void;
+    private selectedItems = new Set<string>();
+    private mappedItems = new Map<string, T>();
+    private submitted = false;
+
+    private SELECTOR_SELECTED = ".is-selected";
+    private CLASS_SELECTED = "is-selected";
+
+    private suggestionsContainer: HTMLDivElement;
+    private chipsContainer: HTMLDivElement;
+    private textInput: TextComponent;
+
+    constructor(
+        private text_items: string[] | ((item: T) => string),
+        private items: T[],
+        private unrestricted: boolean,
+        placeholder: string,
+        title: string,
+        private limit?: number
+    ) {
+        super(app);
+
+        // If `T` is of type `string`, then `unrestricted` is set to provided value
+        // else `unrestricted` is always false
+        if (items.length === 0 || typeof items[0] === "string") {
+            this.unrestricted = unrestricted;
+        }
+
+        // Map each item to its text representation
+        items.forEach((item) =>
+            this.mappedItems.set(this.getItemText(item), item)
+        );
+
+        // Create Elements
+        this.titleEl.setText(title);
+        this.chipsContainer = this.contentEl.createDiv();
+        this.textInput = new TextComponent(this.contentEl);
+        this.textInput.setPlaceholder(placeholder);
```

Comment:
Not entirely sure this needs a placeholder. What use case do you have in mind for it?

Reply:
Since `tp.system.suggester` have placeholder so I thought it would be nice to have an option in `tp.system.multiselect` also

Reply:
Yes, I think it should be removed

Reply:
Should I remove option to set custom title as well?